meetings/minutes/2022-08-01-TSC-Minutes.md

```diff
@@ -0,0 +1,183 @@
+Additionally, it has far better moderation tools; in fact, we can even require users to verify they've read our code of conduct before they can complete registration.
+It does, however, miss a few other features that Slack gives us (threading and reminders in particular).
+
+I see a few different routes we could go:
+
+- Keep Slack.
+  If we do, we'll need to remember that anything older than 90 days disappears.
+  We'll also need a Greetbot replacement (best one I've found with a free plan is [CommonRoom](https://www.commonroom.io/pricing/)).
+- Migrate to Discord.
+  If we do this, we'll need to setup our welcome message templates (I already have them from Greetbot), and:
+  - Update our GitHub webhook on laminas.dev to talk to Discord.
+  - Somehow migrate our users to the new system. (We'd need a plan.)
+
+I'd like to begin discussions, with a goal of, at the minimum, determining how we will address welcome messages when Greetbot closes doors.
+
+#### Discussion
+
+Matthew W apologized for opening this can of worms, and suggested we focus only on the GreetBot replacement.
+When pressed, he suggested that we save discussion of Slack alternatives to after we've brainstormed what we need out of a chat solution, and then done research to determine what options there are, and how each does or does not address those needs.
+
+(This did not stop the inevitable MS Teams bashing, nor the recommendations of replacements!)
+
+#### Decision
+
+We decided to move forward with CommonRoom for the time being.
+
+### Coding style new major release
+
+In the laminas/laminas-coding-standard#75 there are BC breaks in the slevomat ruleset dependency. A rule rename affected us and there are a few more breaks, which do not affect us. Since we (should) lock to minor versions `~2.3.0`, it does not make a difference if we release 2.4 or 3.0 as we (or renovate) need to update all packages either way.
+
+To prevent upgrade issues and supporting multiple versions, the preference would be create a 3.0 release.
+
+#### Discussion
+
+The impetus for this is two items:
+
+- We've observed rulesets change/rename in minor versions, which can cause breakage in our CI.
+- CodeSniffer itself has had breakage even in patch versions (though, as Luís notes, it's rare... but when it happens, it can be painful).
+
+Matthew W suggested that we probably should have a different rule for laminas-coding-standard with regard to new major releases, as any addition or rule change is essentially a BC break; as such, new major releases should be normalized for the tool.
+This was met with agreement.
+
+Max and Geert wondered if we should be pinning the _dependencies_ for laminas-coding-standard to minor versions as well, to prevent breakage.
+
+#### Decision
+
+We decided to roll a new major, and start pinning dependencies to minor releases to prevent future headaches.
```

Comment:
Same here. Decision was to pin to patch versions.

Reply:
for PHPCS itself, not really for everything.

Reply:
Clarifying that point in the minutes; thanks!